spec/ios_l10n_helper_spec.rb

```diff
@@ -0,0 +1,162 @@
+      Dir.mktmpdir('a8c-release-toolkit-l10n-helper-tests-') do |tmp_dir|
+        output_file = File.join(tmp_dir, 'output.strings')
+        expect do
+          described_class.merge_strings(paths: paths, output_path: output_file)
+        end.to raise_exception(RuntimeError, "The file `#{paths[1]}` is in xml format but we currently only support merging `.strings` files in text format.")
+      end
+    end
+
+    it 'raises if one of the strings file does not exist' do
+      paths = [fixture('Localizable-utf16.strings'), '/these/are/not/the/droids/you/are/looking/for']
+      Dir.mktmpdir('a8c-release-toolkit-l10n-helper-tests-') do |tmp_dir|
+        output_file = File.join(tmp_dir, 'output.strings')
+        expect do
+          described_class.merge_strings(paths: paths, output_path: output_file)
+        end.to raise_exception(RuntimeError, "The file `#{paths[1]}` does not exist or is of unknown format.")
+      end
+    end
+  end
+
+  describe '#read_strings_file_as_hash' do
+    it 'can read the content of a textual strings file' do
+      file = fixture('Localizable-utf16.strings')
+      expected_hash = { 'key1' => 'string 1', 'key2' => 'string 2️⃣', 'key3' => 'string 3' }
+      expect(described_class.read_strings_file_as_hash(path: file)).to eq(expected_hash)
+    end
+
+    it 'can read the content of an XML strings file' do
+      file = fixture('xml-format.strings')
+      expected_hash = { 'key1' => 'string 1 xml', 'key2' => 'string 2️⃣ xml', 'key3' => 'string 3 xml' }
+      expect(described_class.read_strings_file_as_hash(path: file)).to eq(expected_hash)
+    end
+
+    it 'can read the content of a binary plist strings file' do
+      file = fixture('bplist-format.strings')
+      expected_hash = { 'key1' => 'string 1 bplist', 'key2' => 'string 2️⃣ bplist', 'key3' => 'string 3 bplist' }
+      expect(described_class.read_strings_file_as_hash(path: file)).to eq(expected_hash)
+    end
+
+    it 'raises if trying to parse an invalid strings file' do
+      file = fixture('not-a-strings-file.png')
+      expect do
+        described_class.read_strings_file_as_hash(path: file)
+      end.to raise_exception(RuntimeError)
+    end
+  end
+
+  describe '#generate_strings_file_from_hash' do
+    it 'generates an XML strings file from a hash' do
+      hash = {
+        InfoKey1: 'Info String 1 translated',
+        InfoKey2: 'Info String 2️⃣ translated',
+        InfoKey3: 'Info String 3 translated'
+      }
+      Dir.mktmpdir('a8c-release-toolkit-l10n-helper-tests-') do |tmp_dir|
+        output_file = File.join(tmp_dir, 'output.strings')
+        expected_file = fixture('expected-generated.strings')
+        described_class.generate_strings_file_from_hash(translations: hash, output_path: output_file)
+        expect(File.read(output_file)).to eq(File.read(expected_file))
+      end
+    end
+
+    # Reads non-latin content from UTF-16BE file and generating output file as UTF8
```

Comment:
Nitpick: verbs consistency

```suggestion
    # Reads non-latin content from UTF-16BE file and generates output file as UTF8
```

or 

```suggestion
    # Reading non-latin content from UTF-16BE file and generating output file as UTF8
```